tests/utils/test_file.py

```diff
@@ -212,3 +213,26 @@ def test_get_modules_from_invalid_file(self):
         modules = list(file_utils.iter_airflow_imports(file_path))
 
         assert len(modules) == 0
+
+
+def test_get_unique_dag_module_name():
+    edge_filenames = [
+        "test_dag.py",
+        "test-dag.py",
+        "test-dag-1.py",
+        "test-dag_1.py",
+        "test-dag.dev.py",
+        "test_dag.prod.py",
+    ]
+    # sha1 of file_path in middle
+    expected_regex = [
+        r"unusual_prefix_[0-9a-f]{40}_test_dag",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_1",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_1",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_dev",
+        r"unusual_prefix_[0-9a-f]{40}_test_dag_prod",
+    ]
```

Comment:
nit. Is it possible to leverage `pytest.mark.parametrize` like this?
```suggestion
@pytest.mark.parametrize(
    "edge_filenames, expected_regex", # sha1 of file_path in middle
    [
        ("test_dag.py", r"unusual_prefix_[0-9a-f]{40}_test_dag"),
        ("test-dag.py", r"unusual_prefix_[0-9a-f]{40}_test_dag"),
        ("test-dag-1.py", r"unusual_prefix_[0-9a-f]{40}_test_dag_1"),
        ("test-dag_1.py", r"unusual_prefix_[0-9a-f]{40}_test_dag_1"),
        ("test-dag.dev.py", r"unusual_prefix_[0-9a-f]{40}_test_dag_dev"),
        ("test_dag.prod.py", r"unusual_prefix_[0-9a-f]{40}_test_dag_prod"),
    ],
)
```

Reply:
Why do we want to check `expected_regex` instead of `modify_module_name`?

Reply:
+1 I think making test on module name is enough in the unit test, not sure @jason810496's thought on this,

Reply:
Since [`get_unique_dag_module_name`](https://github.com/apache/airflow/blob/main/airflow/utils/file.py#L357) returns a unique module name in the format `unusual_prefix_{sha1 of module's file path}_{original module name}`, and I can't mock the path hash since the hash part is coupled within `get_unique_dag_module_name`, I used a regex to validate the expected result.

Reply:
Without using regex to match the expected output, the test will be like this:
```python
@pytest.mark.parametrize(
    "edge_filename, expected_modification_end",
    [
        ("test_dag.py", "_test_dag"),
        # ...
    ],
)
def test_get_unique_dag_module_name(edge_filename, expected_modification_end):
    modify_module_name = file_utils.get_unique_dag_module_name(edge_filename)
    assert modify_module_name.startswith("unusual_prefix_")
    assert modify_module_name.endswith(expected_modification_end)
```

From my perspective, using regex for matching will make it clearer what the expected output of [`get_unique_dag_module_name`](https://github.com/apache/airflow/blob/main/airflow/utils/file.py#L354-L359) is.